pkg/performance/registry.go

```diff
@@ -0,0 +1,107 @@
+// Copyright Antimetal, Inc. All rights reserved.
+//
+// Use of this source code is governed by a source available license that can be found in the
+// LICENSE file or at:
+// https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt
+
+package performance
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+)
+
+type CollectorRegistry struct {
+	pointCollectors      map[MetricType]PointCollector
+	continuousCollectors map[MetricType]ContinuousCollector
+	logger               logr.Logger
+}
+
+func NewCollectorRegistry(logger logr.Logger) *CollectorRegistry {
+	return &CollectorRegistry{
+		pointCollectors:      make(map[MetricType]PointCollector),
+		continuousCollectors: make(map[MetricType]ContinuousCollector),
+		logger:               logger.WithName("registry"),
+	}
+}
+
+func (r *CollectorRegistry) RegisterPoint(collector PointCollector) error {
+	if collector == nil {
+		return fmt.Errorf("cannot register nil collector")
+	}
+
+	metricType := collector.Type()
+	if _, exists := r.pointCollectors[metricType]; exists {
+		return fmt.Errorf("point collector for metric type %s already registered", metricType)
+	}
+	if _, exists := r.continuousCollectors[metricType]; exists {
+		return fmt.Errorf("continuous collector for metric type %s already registered", metricType)
+	}
+
+	r.pointCollectors[metricType] = collector
+	r.logger.Info("registered point collector", "type", metricType, "name", collector.Name())
+	return nil
+}
+
+func (r *CollectorRegistry) RegisterContinuous(collector ContinuousCollector) error {
+	if collector == nil {
+		return fmt.Errorf("cannot register nil collector")
+	}
+
+	metricType := collector.Type()
+	if _, exists := r.continuousCollectors[metricType]; exists {
+		return fmt.Errorf("continuous collector for metric type %s already registered", metricType)
+	}
+	if _, exists := r.pointCollectors[metricType]; exists {
+		return fmt.Errorf("point collector for metric type %s already registered", metricType)
+	}
+
+	r.continuousCollectors[metricType] = collector
+	r.logger.Info("registered continuous collector", "type", metricType, "name", collector.Name())
+	return nil
+}
```

Comment:
Hrmm... I think we are missing that we might have different collector instances? E.g. We might have some collector for each container on a machine, but they would have the same metricType.

Reply:
I just moved the registry stuff to a separate file without changes. Going to follow up on refactoring the registry in a separate PR!